materialbarcodescanner/src/main/java/com/edwardvanraak/materialbarcodescanner/MaterialBarcodeScanner.java

```diff
@@ -64,42 +67,62 @@ public void onBarcodeScannerResult(Barcode barcode) {
      */
     public void startScan() {
         EventBus.getDefault().register(this);
-        if (mMaterialBarcodeScannerBuilder.getActivity() == null) {
+        if (materialBarcodeScannerBuilder.getActivity() == null) {
             throw new RuntimeException("Could not start scan: Activity reference lost (please rebuild the MaterialBarcodeScanner before calling startScan)");
         }
-        int mCameraPermission = ActivityCompat.checkSelfPermission(mMaterialBarcodeScannerBuilder.getActivity(), Manifest.permission.CAMERA);
+        int mCameraPermission = ActivityCompat.checkSelfPermission(materialBarcodeScannerBuilder.getActivity(), Manifest.permission.CAMERA);
         if (mCameraPermission != PackageManager.PERMISSION_GRANTED) {
             requestCameraPermission();
         }
         else {
             //Open activity
             EventBus.getDefault().postSticky(this);
-            Intent intent = new Intent(mMaterialBarcodeScannerBuilder.getActivity(), MaterialBarcodeScannerActivity.class);
-            mMaterialBarcodeScannerBuilder.getActivity().startActivity(intent);
+
+            // new code - attempting to start a fragment
+            FragmentManager fragmentManager = ((AppCompatActivity) materialBarcodeScannerBuilder.getActivity()).getSupportFragmentManager();
+            materialBarcodeScannerFragment = MaterialBarcodeScannerFragment.newInstance(null);
+
+            int id = materialBarcodeScannerBuilder.getRootView().getId();
+
+            fragmentManager.beginTransaction()
+                .replace(id, materialBarcodeScannerFragment)
+                .addToBackStack(MaterialBarcodeScannerFragment.class.getSimpleName())
```

Comment:
There's actually no need to, but this method is overridden by a class in the tru-dl sdk in which we don't add the fragment to the back stack.